ipld/cid/src/lib.rs

```diff
@@ -43,7 +42,7 @@ pub struct Prefix {
 }
 
 /// Representation of a IPLD CID.
-#[derive(Eq, Clone)]
+#[derive(PartialEq, Eq, Hash, Clone)]
```

Comment:
It was intentional to keep those separate before (this was a fork of existing implementations initially) and the specific details of comparing and hashing bytes was necessary. I can't see at this point for that specific comparison (cids aren't used as Hamt keys and that was going to be the only potential need to be specific).

Reply:
Hmm, could could two CIDs ever be considered equal in one way but not in the other?

Reply:
Equality, no, but ordering potentially yes because of the variant encoding. Just something to keep in mind in the future, that would be a very nuanced bug to find if there was an inconsistency

Reply:
So are you talking about ordering and not about `Eq` and `Hash`? I agree that changing the ordering can change behavior but here I'm getting rid of `Cid`'s `Ord` implementation altogether, rather than changing it. Deriving `Eq` and `Hash` should not result in observable changes as long as the implementation is correct

Reply:
hash will be different too, only PartialEq/Eq would be the same (sorry for bringing up order, was just looking on my phone)

Reply:
But I don't think we use the cid hash for any consensus related things so should be fine

Reply:
Ah yeah, the hash will be different, but as far as I understand you would never look at the resulting hash value anyway, or somehow rely on what it is – i.e. previously the hash of a cid was the hash of its bytes which as far as we're concerned could be anything

Reply:
Yeah, I mean it could matter if you use a cid as the key to a hamt, but that shouldn't be a need in the near future because the go implementation statically uses bytes for their keys

Reply:
I will definitely keep this in mind for if we're ever going to use the CID as the key to a hamt